weaveworks/eksctl
# Allow additional volume mappings

Closes https://github.com/weaveworks/eksctl/issues/2115

Todo: 

Probably some better docs.

## Manual Test

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: test-additional-volumes-2
  region: us-west-2
  version: '1.20'

managedNodeGroups:
  - name: managed-ng-1
    minSize: 1
    maxSize: 4
    desiredCapacity: 1
    additionalVolumes:
      - volumeName: '/dev/sdb'
```

```
➜  additional-volumes eksctl create cluster -f cluster-managed.yaml
...
2022-02-01 14:39:50 [ℹ]  nodegroup "managed-ng-1" has 1 node(s)
2022-02-01 14:39:50 [ℹ]  node "ip-192-168-76-90.us-west-2.compute.internal" is ready
2022-02-01 14:39:52 [ℹ]  kubectl command should work with "/Users/skarlso/.kube/config", try 'kubectl get nodes'
2022-02-01 14:39:52 [✔]  EKS cluster "test-additional-volumes-2" in "us-west-2" region is ready
```

<img width="704" alt="Screenshot 2022-02-01 at 14 43 09" src="https://user-images.githubusercontent.com/182850/151979240-e59747ab-e34e-4162-ba18-7d3c6b59aa2d.png">


TODO: 

Update the acceptance test

Currently running acceptance test....

```
using API server https://7D1D98844D284ED6A487D4A8270C7A95.gr7.us-west-2.eks.amazonaws.com

• [SLOW TEST:350.853 seconds]
(Integration) Create Managed Nodegroups
/Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/managed/managed_nodegroup_test.go:57
  cluster with 1 al2 managed nodegroup
  /Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/managed/managed_nodegroup_test.go:159
    supports adding bottlerocket and ubuntu nodegroups with additional volumes
    /Users/skarlso/goprojects/weaveworks/eksctl/integration/tests/managed/managed_nodegroup_test.go:160
....
Ran 1 of 21 Specs in 1665.693 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 20 Skipped
PASS | FOCUSED
```

Same for running unmanaged nodegroups.

```
Ran 1 of 43 Specs in 2362.203 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 42 Skipped
PASS | FOCUSED
```
